src/nevermined/api/AssetsApi.ts

```diff
@@ -0,0 +1,464 @@
+        return this.registerNeverminedAsset(
+            assetAttributes,
+            publisher,
+            publishMetadata,
+            undefined,
+            txParams
+        )
+    }
+
+
+    /**
+     * Given a DID, updates the metadata associated to the asset. It also can upload this metadata to a remote decentralized stored depending on the `publishMetadata` parameter.
+     * 
+     * @example
+     * ```ts
+     * const ddoUpdated = await nevermined.assets.update(
+     *      ddo.shortId(), 
+     *      updatedMetadata, 
+     *      publisher, 
+     *      PublishMetadata.IPFS
+     * )
+     * ```
+     * 
+     * @param did - Decentralized ID representing the unique id of an asset in a Nevermined network.
+     * @param metadata - Metadata describing the asset
+     * @param publisher - Account of the user updating the metadata
+     * @param publishMetadata - It allows to specify where to store the metadata  
+     * @param txParams - Optional transaction parameters
+     * @returns {@link DDO} The DDO updated
+     */
+     public update(
+        did: string,
+        metadata: MetaData,
+        publisher: Account,
+        publishMetadata: PublishMetadata = PublishMetadata.OnlyMetadataAPI,
+        txParams?: TxParameters
+    ): SubscribablePromise<UpdateProgressStep, DDO> {
+        return this.updateAsset(did, metadata, publisher, publishMetadata, txParams)
+    }
+
+
+    /**
+     * Start the purchase/order of an access service. Starts by signing the service agreement
+     * then sends the request to the publisher via the service endpoint (Node http service).
+     * If the access service to purchase is having associated some price, it will make the payment
+     * for that service.
+     * @param did - Unique identifier of the asset to order
+     * @param consumerAccount - The account of the user ordering the asset
+     * @param txParams - Optional transaction parameters
+     * @returns The agreement ID identifying the order
+     */
+     public order(
+        did: string,
+        consumerAccount: Account,
+        txParams?: TxParameters
+    ): SubscribablePromise<OrderProgressStep, string> {
+        return this.orderAsset(did, 'access', consumerAccount, txParams)
+    }
+
+
+    /**
+     * Having previously ordered an "access" service (referenced via an "agreementId"). 
+     * This method allows to download the assets associated to that service.
+     * @param agreementId  - The unique identifier of the order placed for a service
+     * @param did - Unique identifier of the asset ordered
+     * @param consumerAccount - The account of the user who ordered the asset and is downloading the files
+     * @param resultPath - Where the files will be downloaded
+     * @param fileIndex - The file to download. If not given or is -1 it will download all of them.
+     * @returns The result path or true if everything went okay
+     */
+    public async consume(
+        agreementId: string,
```

Comment:
do we really need to keep using exposing agreementId or should the gateway be clever enough to find it?

Reply:
I think we need it. Technically the same consumer could have 2 orders for the same asset, so the agreementId helps to specify which one the consumer refers to